### DISCUSSION TITLE
Error: `Conversion from Uuid() to INT failed` on findMany()

### DISCUSSION THREAD

**Discussion by emmanuel-londono** [prisma/prisma#21476] [Q&A] [answered]
### Question

Why am I receiving the following error when I visit the endpoint  /users, and what are possible fix routes? 

```js
[1]   19 const usersRouter = express_1.default.Router();
[1]   20 usersRouter.get("/", (req, res) => __awaiter(void 0, void 0, void 0, function* () {
[1]   21     try {
[1] → 22         const allUsers = yield prisma.user.findMany(
[1] Error in domain logic: Conversion from `Uuid(ac44955b-ea51-474a-ae89-5a6e1744df9a)` to `Int` failed.
[1]     at Cn.handleRequestError (~\backend\node_modules\@prisma\client\runtime\library.js:123:6989)
[1]     at Cn.handleAndLogRequestError (~\backend\node_modules\@prisma\client\runtime\library.js:123:6206)
[1]     at Cn.request (~\backend\node_modules\@prisma\client\runtime\library.js:123:5926)
[1]     at async l (~\backend\node_modules\@prisma\client\runtime\library.js:128:9968)
```

### Expected Behavior
Whenever I visit the endpoint /users, I should be responded with a JSON of all of our users.

```js
[
  {
    "userid": "d536eb96-02d5-4a7b-8b72-90c1585db30d",
    "user_name": "mockuser1",
    "user_abbreviation": "mu1"
  },
  {
    "userid": "35c709e3-b498-4493-856b-f46b8df370a1",
    "user_name": "mockuser2",
    "user_abbreviation": "mu2"
  },
 {
    "userid": "99718d23-bd15-446a-aef7-bb71c844bec5",
    "user_name": "mockuser3",
    "user_abbreviation": "mu3"
  },
]
```

### How to reproduce 

Background information:

Backend Technologies in use: [Node, Express, Typescript, Prisma, Postgres]

I'm at an early stage in our prisma implementation.

I'm running a postgres database which uses UUID strings as primary keys for each table. The database was formerly built out (seeded) in postgres.

Used `npx prisma db pull` to successfully create a prisma schema .

Currently, expressJS is setup to run prisma commands and perform CRUD functionalities.

```js
import express from "express";
import { PrismaClient } from "@prisma/client";
import { busyHandler, catchHandler, handleError } from "../../common/common.error";
const prisma = new PrismaClient();
const usersRouter = express.Router();

usersRouter.get("/", async (req, res) => {
    try {
        const allUsers = await prisma.user.findMany();
        res.send(allUsers);
    } catch (err) {
        catchHandler(res, err);
    } finally {
        await prisma.$disconnect();
    }
});
```
### What I've tried

I've also tried the following:

- replacing the respond method in the try-catch to res.json(allUsers):
```js
usersRouter.get("/", async (req, res) => {
    try {
        const allUsers = await prisma.user.findMany();
        res.json(allUsers);
    } catch (err) {
        catchHandler(res, err);
    } finally {
        await prisma.$disconnect();
    }
});
```
- filtering the findMany:
```js
usersRouter.get("/", async (req, res) => {
    try {
        const user = await prisma.user.findMany(where:{userid:'ac44955b-ea51-474a-ae89-5a6e1744df9a');
        res.send(user);
    } catch (err) {
        catchHandler(res, err);
    } finally {
        await prisma.$disconnect();
    }
});
```

### Information about Prisma Schema, Client Queries and Environment 

```prisma
model user {
  userid                       String           @id @db.Uuid
  user_name              String?
  user_abbreviation      String?
}
```

```js
usersRouter.get("/", async (req, res) => {
    try {
        const allUsers = await prisma.user.findMany();
        res.send(allUsers);
    } catch (err) {
        catchHandler(res, err);
    } finally {
        await prisma.$disconnect();
    }
});
```

- OS: Windows
- Database: PostgreSQL
- Node.js version: v16.14.2
- RestAPI: Expressjs
- Language: Typescript
- prisma version: 5.3.1
- @prisma/client version: 5.4.2

**Comment by ludralph** [comment]
Hi @emmanuel-londono 👋 

Thank you for raising this question. 

Firstly, can you please ensure that both the Prisma version and the Prisma Client version are the same?  I can see you are using Prisma client version 5.4.2 and the Prisma version 5.3.1. Can you ensure both of them are 5.4.2? Also, can you share the GitHub repository so I can investigate this?

**Reply by emmanuel-londono** [reply]
Hi @ludralph, I've updated prisma to 5.4.2 per your advice. I tried to fetch that endpoint again but still get the same error.

This project's codebase is unfortunately hosted in gitlab and it's privatized, but I will try and make a small github project to replicate this functionality when I have the time.

**Reply by emmanuel-londono** [reply]
I was looking through the index.d.ts file to see if there was a way to not convert to int but was unable to find it.

In the past, I used integers as my primary keys for the postgres tables. Our early Prisma implementation worked, no problem. It was when I switched to using UUIDs in the database that caused the hiccup and reason for the ticket.

Would it be preferable if we switch back to integers for our primary keys?

**Reply by ludralph** [reply, marked as answer]
@emmanuel-londono The error message you're seeing suggests that somewhere in your application or database, a `UUID` is being incorrectly treated as an `Int`. However, from the code and schema you've provided, it's not clear where this could be happening.
The error could be coming from another part of your application that's not shown in the code you've provided. Check any other code that interacts with the user table and ensure it's treating `userid` as a String.
